pkg/platform/runtime/ARCHITECTURE.md

```diff
@@ -9,11 +9,15 @@ The general usage pattern is as follows:
 
 	rt, err := runtime.New(target)
 	if err != nil {
-		if !runtime.IsNeedsUpdateError(err) {
-			return err
-		}
-		if err = rt.Update(messageHandler); err != nil {
-			return err
+		switch {
+			case runtime.IsNeedsUpdateError(err):
+				if err = rt.Update(messageHandler); err != nil {
+					return err
+				}
+			case runtime.IsNeedsStageError(err):
+				...
+			default:
+				return err
```

Comment:
Nit: Maybe slightly cleaner (this assumes dropping the `if` on line 11):

```
		switch {
			case err == nil:
				break
			case runtime.IsNeedsUpdateError(err):
				if err = rt.Update(messageHandler); err != nil {
					return err
				}
			case runtime.IsNeedsStageError(err):
				...
			default:
				return err
 		}
```

Reply:
I like it! Thanks.